filters.go

```diff
@@ -311,6 +311,38 @@ func (p *Pipe) MatchRegexp(re *regexp.Regexp) *Pipe {
 	})
 }
 
+// ReadInput reads displays the pipe content as a message as a prompt
+// and reads the content from standard input and returns the result
+// as a pipe. In case the input was empty it defaults to its `defaultValue`
+func (p *Pipe) ReadInput(defaultValue string) *Pipe {
+	const emptyString = ""
+	const withDefault = "%s [%s]: "
```

Comment:
I don't think these constants are useful either; in fact, they make the code obscure, because when you're looking at line 331, for example, you can't see what `withDefault` actually is. You have to jump elsewhere in the source to find it and then jump back. That's a speed bump for the reader.

Reply:
Fair point.

However, I think that given the relative small size of the function body; I'd say defining constants is good actually for readability — the maintainer doesn't really need to know what the actual strings are. Plus if we have to reuse the constants they're easy-to-find and to change.

Defining `withDefault` and `withoutDefault` up-first; gives a good contrast as to what they semantically represent.

If you still think it's better to use plain literals; I'm happy to do the change.

Reply:
Well, if we define the constants right next to where they're used, it's just as easy to find and change the _literals_ instead, isn't it?

The only real value in defining a constant is so that it can be a long way away from the place where it's used. But we _don't_ want that here.

Reply:
IMHO, this discussion seems to not have an objective answer. I'm more than glad to follow the project conventions and use the literals directly.